### PR TITLE
Uat admin role

### PR DESCRIPTION
**Pull request by EvanBarbour3**
## Features
- Location reference validation added
- Locations are now added as active initially

## Bug Fixes
- Edit forms changed to look the same and display correctly
- Breadcrumbs edited to work on smaller screens
- Action buttons now go smaller when on smaller devices
- Active toggle being consistently used
- Login page tabbing changed to work correctly
- Navbar fixes

[Delivers#102846772]

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 0.13.0
```
Comment (wnowicki) [inline CHANGELOG.md]: What ?

File: app/Http/Controllers/LocationsController.php [hunk 2]
```diff
@@ -64,13 +64,12 @@ public function create()
     public function store(Request $request)
     {
         $this->validate($request, [
+            'reference' => 'required|regex:/^[A-Za-z0-9\-!]+$/',
```
Comment (wnowicki) [inline app/Http/Controllers/LocationsController.php]: Sure about `[A-Za-z0-9\-!]` ??

File: app/Http/Controllers/LocationsController.php [hunk 3]
```diff
@@ -123,10 +122,11 @@ public function edit($id)
     public function update($id, Request $request)
     {
         $this->validate($request, [
+            'reference' => 'required|regex:/^[A-Za-z0-9\-!]+$/',
```
Comment (wnowicki) [inline app/Http/Controllers/LocationsController.php]: Can we change reference at all?

File: app/Http/Controllers/MerchantsController.php [hunk 4]
```diff
@@ -122,7 +123,8 @@ public function update($id, Request $request)
     {
         $this->validate($request, [
             'name' => 'required',
-            'token' => 'required'
+            'token' => 'required',
+            'active' => 'required',
```
Comment (wnowicki) [inline app/Http/Controllers/MerchantsController.php]: is not required to be active

File: database/migrations/2015_06_30_141244_basic_models.php [hunk 5]
```diff
@@ -53,7 +53,7 @@ public function up()
             $table->increments('id');
             $table->string('reference');
             $table->integer('installation_id')->unsigned();
-            $table->boolean('active');
+            $table->boolean('active')->default(true);
```
Comment (wnowicki) [inline database/migrations/2015_06_30_141244_basic_models.php]: What ???!!!

**Comment by wnowicki** [comment]
@EvanBarbour3 please review my comments and then we can go through frontend together

File: app/Http/Controllers/LocationsController.php (hunk 2 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/LocationsController.php]: Accidental '!', has been removed

File: app/Http/Controllers/LocationsController.php (hunk 3 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/LocationsController.php]: Nope, removed reference from the edit page on locations, and removed the validation from the controller

File: app/Http/Controllers/MerchantsController.php (hunk 4 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/MerchantsController.php]: Okay, I've removed this

File: database/migrations/2015_06_30_141244_basic_models.php (hunk 5 above)
Comment (EvanBarbour3) [inline database/migrations/2015_06_30_141244_basic_models.php]: Added into a new migration

**Comment by wnowicki** [comment]
@EvanBarbour3 is this PR ready for the review?

**Comment by EvanBarbour3** [comment]
@wnowicki Yes

File: app/Http/Controllers/Controller.php [hunk 6]
```diff
@@ -378,7 +379,7 @@ protected function checkModelForMerchantLimit(Model $entity, $merchantId, $model
 
     /**
      * @author EB
-     * @return Carbon[]
+     * @return integer
```
Comment (wnowicki) [inline app/Http/Controllers/Controller.php]: I think it's still `Carbon[]`

File: app/Http/Controllers/Controller.php [hunk 7]
```diff
@@ -476,10 +477,10 @@ protected function fetchFilterValues($model, $filter)
 
     /**
      * @author EB
-     * @param $model
-     * @param $filter
-     * @param $false
-     * @param $true
+     * @param Builder $model
+     * @param string $filter
+     * @param string $false
```
Comment (wnowicki) [inline app/Http/Controllers/Controller.php]: `$true` and `$false` are really bad names for params... especially if you want them to be type of `string`

File: app/Http/Controllers/Controller.php [hunk 8]
```diff
@@ -497,8 +498,8 @@ protected function fetchBooleanFilterValues($model, $filter, $false, $true)
 
     /**
      * @author EB
-     * @param $model
-     * @param $associate
+     * @param Builder $model
+     * @param string $associate
      * @return array
      */
     protected function fetchAssociateFilterValues($model, $associate)
```
Comment (wnowicki) [inline app/Http/Controllers/Controller.php]: shouldn't be that an optional param?

File: app/Http/Controllers/InstallationsController.php [hunk 9]
```diff
@@ -101,14 +102,18 @@ public function edit($id)
     public function update($id, Request $request)
     {
         $this->validate($request, [
+            'name' => 'required',
             'validity' => 'required|integer|between:7200,604800',
+            'custom_logo_url' => 'url',
+            'ext_return_url' => 'url',
+            'ext_notification_url' => 'url',
+
```
Comment (wnowicki) [inline app/Http/Controllers/InstallationsController.php]: don't need empty line here

File: database/migrations/2015_09_25_130543_add_default_to_active_on_locations.php [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        //
```
Comment (wnowicki) [inline database/migrations/2015_09_25_130543_add_default_to_active_on_locations.php]: No `down()` ?

**Comment by wnowicki** [comment]
@EvanBarbour3 some minor thing to fix before we can review the frontend

File: app/Http/Controllers/Controller.php (hunk 6 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/Controller.php]: Changed back to Carbon[]

File: app/Http/Controllers/Controller.php (hunk 7 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/Controller.php]: Changed to $falseLabel and $trueLabel

File: app/Http/Controllers/InstallationsController.php (hunk 9 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/InstallationsController.php]: Removed empty line

File: database/migrations/2015_09_25_130543_add_default_to_active_on_locations.php (hunk 10 above)
Comment (EvanBarbour3) [inline database/migrations/2015_09_25_130543_add_default_to_active_on_locations.php]: Added down() to change it to false again

File: app/Http/Controllers/Controller.php (hunk 8 above)
Comment (EvanBarbour3) [inline app/Http/Controllers/Controller.php]: It's fine, just been over this

**Comment by EvanBarbour3** [comment]
Ready for review

File: app/Http/Controllers/Controller.php [hunk 11]
```diff
@@ -526,4 +527,18 @@ protected function fetchApplicationDetails($id)
     {
         return Application::where('ext_id', '=', $id)->first();
     }
+
+    /**
+     * @author EB
+     * @param $request
+     * @return array
+     */
+    protected function updateWithCheckboxes($request)
```
Comment (wnowicki) [inline app/Http/Controllers/Controller.php]: 1. Bad name
2. What is the type of `$request` ??
3. `$request->toArray()` or something similar?

Comment (wnowicki) [inline app/Http/Controllers/Controller.php]: replace this with `array_values()`

**Pull request merged**